sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/StoreReader.java

```diff
@@ -656,6 +649,32 @@ private ReadReplicaResult createReadReplicaResult(List<StoreResult> responseResu
         return task;
     }
 
+    StoreResult createAndRecordStoreResult(
+        RxDocumentServiceRequest request,
+        StoreResponse storeResponse,
+        Exception responseException,
+        boolean requiresValidLsn,
+        boolean useLocalLSNBasedHeaders,
+        Uri storePhysicalAddress) {
+
+        StoreResult storeResult = this.createStoreResult(storeResponse, responseException, requiresValidLsn, useLocalLSNBasedHeaders, storePhysicalAddress);
+
+        try {
+            BridgeInternal.recordResponse(request.requestContext.cosmosDiagnostics, request, storeResult);
+            if (request.requestContext.requestChargeTracker != null) {
+                request.requestContext.requestChargeTracker.addCharge(storeResult.requestCharge);
+            }
+        } catch (Exception e){
+            logger.error("Unexpected failure while recording response", e);
+        }
+
+        if (responseException !=null) {
+            verifyCanContinueOnException(storeResult.getException());
```

Comment:
There is one problem observed if throw exception here: based on the current logic, we may do not record the response from other replicas:
 List<Flux<StoreResult>> storeResult = readStoreTasks
                .stream()
                .map(item -> toStoreResult(entity, item, readMode, requiresValidLsn))
                .collect(Collectors.toList());
        Flux<StoreResult> allStoreResults = Flux.merge(storeResult);
        return allStoreResults.collectList().onErrorResume(e -> {


Two solutions can think of now:
1. do not do verifyCanContinueOnException check when create and record store results, add the logic after get all store results
2. Instead of Flux.merge, change to Flux.mergeDelayError

Reply:
I think its fine to continue with current logic , because we are not interested in result of other replica in case of these exceptions and break the flow

Reply:
Good observation: I would prefer option 2) - but am not quite sure yet whether there are certain scenarios - like 429 from only a subset of replicas - where this could negatively impact latency. My assumption is that the impact is negligible and 2) looks cleaner to me - so my vote would be for 2)

@simplynaveen20  / @moderakh - what do you think?

Reply:
We don't need either of these two options, earlier code was also not caring about other replica responses in case of certain exceptions. Its like fire and forget if we hit error in any replica , we don't care about the response from other replica. In case of multiple responses come successfully  together before this code , we will have multiple store result in diagnostics , which is also fine.
So i don't think any code change is needed and above two options have unnecessary perf implication . Have offline conversation with Annie on this